File: jscript/j00.arguments.js
var arg_count   = WScript.Arguments.Count() // get num of arguments
var script_name = WScript.ScriptName      // get script name


// check for only two arguments
if (WScript.Arguments.Count() != 2) {
 // print helpful instructions
 WScript.Echo("\nYou need to enter two numbers:\n");
 WScript.Echo("   Usage: " + script_name + " [num1] [num2]");
} else {
 // get sum of both arguments, force cast to integer
 var sum = parseInt(WScript.Arguments.Item(0)) + parseInt(WScript.Arguments.Item(1));
 // print results of both arguments (string) and summation (integer)
 WScript.Echo("The sum of " + WScript.Arguments.Item(0) + 
              " and " + WScript.Arguments.Item(1) + 
              " is: " + sum
             );
}

File: scripts/ruby/j00.arguments.rb
#!/usr/bin/env ruby

arg_count   = ARGV.length # get num of args
script_name = $0          # get script name

# check for only 2 arguments
if arg_count != 2 then
   # output usage statement to standard error
   $stderr.puts "\nYou need to enter two numbers: \n\n"
   $stderr.puts "   Usage: #{script_name} [num1] [num2]\n\n" 
else
   # get sum of both arguments, force cast them to integer
   sum = ARGV[0].to_i + ARGV[1].to_i
   # print results of both arguments (string) and summation (integer)
   puts "The sum of #{ARGV[0]} and #{ARGV[1]} is: #{sum}"
end
